vignettes/sentimentanalyzerR.Rmd

```diff
@@ -0,0 +1,102 @@
+
+Wordcloud - A condensed representation of selected words having the highest influence towards a particular sentiment category. The size of the text is relative to their frequency of occurrence in the document/corpus.
+
+## Functions
+
+This section provides a walk-through of the package functions. A sample dataset is included as a use case for demonstration purposes.
+
+`get_aggregated_sentiment_score` - Computes the majors sentiment of the population.
+
+`likert_scale` - Converts the sentiment score to a likert scale ranging from 1-5.
+
+`sentiment_score_plot` - Creates a binned histogram showing count of reviews against the sentiment score.
+
+`generate_wordcloud` - Creates wordcloud of the most common positive, negative and neutral words.
+
+### Use Case
+
+One of the popular Use Cases for sentiment analysis would be understanding the general sentiment of tweets after a major event. For the demonstration of this Use case, we will utilize the tweets data set.
+
+```{r data, eval=TRUE, message = FALSE}
+# Loading the library and selecting the top 200 rows for the demonstration
+library(readr)
+df <- read_csv("../tests/test_tweets.csv")
+df <- head(df, 200)
+```
+
+### aggregate_sentiment_score
+
+`aggregate_sentiment_score` computes the major sentiment of the corpus of the dataset provide. This represents the average of the sentiment score for each tweet in the dataset.
+
+```{r aggregate-score, eval = TRUE}
+score <- aggregate_sentiment_score(df, "text")
+
+# view the computed score
+score
+```
+
+### likert_scale
+
+The output from the previous function is not very interpretable as it is spread across a sentiment scale of -1 to +1. To make this understandable to the common user, call the `convert_to_likert` function as below.
+
+```{r likert, eval = TRUE, }
+likert_score <- likert_scale(df, "text")
+
+# view the likert scale score
+likert_score
+```
+
+This returns a the overall score on the likert . The return value of 3 corresponds to the "Neutral" sentiment as per the Likert scale mentioned in the function documentation.
+
+### sentiment_score_plot
+
+While understanding the overall sentiment is essential, a more insightful representation can be obtained through a histogram depicting the most frequent sentiments seen in the corpus. Each count represents a unique document in the corpus. Thus the plot can communicate how many reviews fall in each range of sentiment scores.
+
+```{r hist-plot, eval = TRUE, results='hide', mesage = FALSE, fig.keep='all'}
+hist_plot <- sentiment_score_plot(df, "text")
+
+```
+
+**TBD** - Scope available to alter code to show the the statement with insights.
```

Comment:
Does this need to be removed?

Reply:
It's like a placeholder. There is a paste statement in the code that prints a summary of the insights, that is not returned at present. Depending on the code changes done I need to update this prior to milestone 4. Is it ok to have placeholder like that